src/nifreeze/viz/bland_altman.py

```diff
@@ -0,0 +1,284 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Bland-Altman plot."""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.stats as stats
+
+
+def _compute_z_score(ci):
+    """Compute the z-score for the given confidence interval value.
+
+    Parameters
+    ----------
+    ci : :obj:`float`
+        Confidence interval value.
+
+    Returns
+    -------
+    :obj:`float`
+        Z-score value.
+    """
+
+    # Compute the z-score for confidence interval (two-tailed)
+    p = (1 - ci) / 2
+    q = 1 - p
+    return stats.norm.ppf(q)
+
+
+def _compute_bland_altman_features(data1, data2, ci):
+    """Compute quantities of interest for the Bland-Altman plot.
+
+    Parameters
+    ----------
+    data1 : :obj:`numpy.ndarray`
+        Data values.
+    data2 : :obj:`numpy.ndarray`
+        Data values.
+    ci : :obj:`float`
+        Confidence interval value.
+
+    Returns
+    -------
+    diff : :obj:`numpy.ndarray`
+        Differences.
+    mean_diff : :obj:`numpy.ndarray`
+        Mean differences.
+    mean : :obj:`numpy.ndarray`
+        Mean values.
+    std_diff : :obj:`numpy.ndarray`
+        Standard deviation of differences.
+    loa_lower : :obj:`numpy.ndarray`
+        Lower limit of agreement.
+    loa_upper : :obj:`numpy.ndarray`
+        Upper limit of agreement.
+    ci_mean : :obj:`numpy.ndarray`
+        Confidence interval of mean values.
+    ci_loa : :obj:`numpy.ndarray`
+        Confidence interval of limits of agreement.
+    """
+
+    # Compute mean
+    mean = np.mean([data1, data2], axis=0)
+
+    # Compute differences, mean difference, and std dev of differences
+    diff = data1 - data2
+    mean_diff = np.mean(diff)
+    std_diff = np.std(diff, axis=0)
+    # std_diff = np.std(diff, ddof=1)  # Use Bessel's correction
+
+    # Compute confidence interval limits of agreement (LoA)
+    z_score = _compute_z_score(ci)
+    loa_lower = mean_diff - z_score * std_diff
+    loa_upper = mean_diff + z_score * std_diff
+
+    n = len(diff)
+
+    # Compute t-value for the confidence intervals (CIs)
+    t_val = stats.t.ppf((1 + ci) / 2, n - 1)
```

Comment:
Might be clearer as: 

```suggestion
    t_val = stats.t.ppf(0.5 + ci / 2, n - 1)
```